docs/basic/flattening.md

```diff
@@ -0,0 +1,134 @@
+
+![tilt removal_better_viewing](../_static/images/flattening/flattening_tilt_removal.png)
+
+## Polynomial removal
+
+After the tilt, we remove the polynomial trends. In some images, there is also quadratic or occasionally cubic bowing to
+the image too. We remove this by fitting a two dimensional quadratic polynomial to the image (in the horizontal
+direction), and subtracting it from the image. We then do the same for a nonlinear polynomial (z = a*x*y) to eliminate
+“saddle” trends in the data. We could do all of these at the same time, but we like to be able to see the iterative
+differences.
+
+## Scar removal
+
+We then run scar removal on the image. This is a special function that detects scars (long, thin, bright streaks in the
+data, caused by physical problems in the AFM process). We are using a different image here since our lovely
+minicircles.spm image doesn’t have any scars.
+
+![scarred image](../_static/images/flattening/flattening_scarred_image.png)
+
+![scar removed](../_static/images/flattening/flattening_scar_removed.png)
+
+**Note that scar removal can distort data, and it’s best to take data without scars if you can.**
+
+## Zero the average height
+
+![height zeroing](../_static/images/flattening/flattening_height_zeroing.png)
+
+We then lower the image by its median height which causes the background of the image to be roughly centred at zero nm.
+This is important since both the AFM and these processing steps can cause your background height to not be zero, which
+would mean you would get the wrong heights for your data.
+
+## Masking
+
+Now consider that all the processing we have done has assumed that every pixel of the image is background. We assumed
+that there were no objects on the surface, messing up our fitting, and row alignment. If there was a large amount of
+DNA on one side of the image, then the slope will be affected by it, and so flatten the image poorly.
+
+Because of this, once we have done our initial flattening, we detect our objects on the surface, and then flatten the
+image again! But this time, ignoring the data on the surface, and only considering the background.
+
+How do we do that?
+Well first, we need to find the data on the surface. We do this by thresholding.
+The threshold is set by the config file (have a look!). Any pixels that are below the threshold, are considered
+background (sample surface). Any pixels that are above the threshold are considered to be data (useful sample objects).
+This binary classification allows us to make a binary mask of where is foreground data, and where is background.
+
+For more information on thresholding and how to set it, see the [thresholding](thresholding.md) page.
+
+Here is the binary mask for minicircle.spm:
+
+![processed image so far](../_static/images/flattening/flattening_tilt_removal.png)
+
+![binary mask](../_static/images/flattening/flattening_binary_mask.png)
```

Comment:
Would it be worth using Illustrator to put these on the same line, mand maybe point out the threshold value on the height scale?